### PR TITLE
Switching back to good old if statement!

### PR DESCRIPTION
**Pull request by hjafarpour**
### Description 
This is a bug fix patch for #1784 .
Switching back to using `if` statement fixes the issue.
Previously we used statements like the following:
```
return node.getLocation()
        .map(location ->
            new ArithmeticBinaryExpression(location, node.getType(),
                (Expression) process(node.getLeft(), context),
                (Expression) process(node.getRight(), context)))
        .orElse(new ArithmeticBinaryExpression(node.getType(),
            (Expression) process(node.getLeft(), context),
            (Expression) process(node.getRight(), context)));
```

This results in execution of both map and orElse arguments which results in exponential visits of the AST nodes.
By changing it to `if` we just run the block that we need.

### Testing done 
_Describe the testing stategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by rodesai** [comment]
Makes sense. ```orElse``` is always going to be called. And its parameter has to be evaluated before being passed.

File: ksql-engine/src/test/resources/query-validation-tests/project-filter.json [hunk 1]
```diff
@@ -43,6 +43,29 @@
         }
       ]
     },
+    {
+      "name": "WHERE with many comparisons",
```
Comment (rodesai) [inline ksql-engine/src/test/resources/query-validation-tests/project-filter.json]: Can you add the issue number, so we know why this test is important?

Comment (hjafarpour) [reply ksql-engine/src/test/resources/query-validation-tests/project-filter.json]: Added the issue number.

**Comment by hjafarpour** [comment]
@rodesai added some comments to the code so future changes don't make the same mistake.

File: ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java [hunk 2]
```diff
@@ -131,67 +134,63 @@ protected Node visitArithmeticBinary(
...
-    return node.getLocation()
-        .map(location ->
-            new ArithmeticBinaryExpression(location, node.getType(),
-                (Expression) process(node.getLeft(), context),
-                (Expression) process(node.getRight(), context)))
-        .orElse(new ArithmeticBinaryExpression(node.getType(),
-            (Expression) process(node.getLeft(), context),
-            (Expression) process(node.getRight(), context)));
-
+    // Don't change this if/else block so only one object gets instanciated.
```
Comment (rodesai) [inline ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java]: - instaciated -> instantiated
- maybe rephrase to something like "use an if/else block here (instead of isPresent.map(...).orElse(...)) so only one object gets instantiated (issue #1784)"

**Comment by ijuma** [comment]
`orElseGet` is lazy, `orElse` is strict

**Comment by apurvam** [comment]
Yes, we discussed this in our team meeting and @big-andy-coates brought up the same point. If the original implementation had used `orElseGet` things would have been fine. Good learning for us all!
